static/js/home.js

```diff
@@ -185,40 +189,57 @@ function getEvents(){
         'facetObj':JSON.stringify(facetObj)
     };
 
-    if(maptToPlot == "roadsegments") {
-        jQuery.post(
-            "/get_segs",
-            bounddic,
-            getEventCB,
-            'json');
-    }else{
-        jQuery.post(
-            "/get_accidents",
-            bounddic,
-            getEventCB,
-            'json');
-    }
+  if(maptToPlot == "roadsegments") {
+    jQuery.post(
+        "/get_segs",
+        bounddic,
+        getEventCB,
+        'json');
+  }else{
+    jQuery.post(
+        "/get_accidents",
+        bounddic,
+        getEventCB,
+        'json');
+  }
+}
+
+function clearVisual(){
+  if(visual_mode == undefined)  return;
+  if(visual_mode === 'markers') {
+    set_markers(onscreenMarker, null); 
+  } else if(visual_mode === 'segments') {
+    setSegments(roadsegments, null);
+  } else {
+    visual_mode.setMap(null); 
+  }
 }
 
 function getEventCB(data){
+  if(lastShape != undefined){
+    lastShape.setMap(null);
+  }
   if(data.status != 0){
     alert(data.data);
     return;
   }
-  if(lastShape == undefined) {
-    set_markers(onscreenMarker, null);
-    onscreenMarker = getAndRenderInitMarkers(map, data.data);
-    document.getElementById('info-box').textContent = onscreenMarker.length + " accident(s) showing in screen"
-  }
+  console.log(maptToPlot);
   acc_list = data.data;
   if(maptToPlot == "heatmap") {
-    render_heatmap(map,acc_list);
+    clearVisual();
+    visual_mode = getAndRenderHeatmap(map, acc_list);
     console.log("rendering the heat map");
   } else if(maptToPlot == "roadsegments") {
-    render_roadsegments(map, acc_list);
+    clearVisual();
+    roadsegments = getAndRenderRoadsegments(map, acc_list);
```

Comment:
Better separate the heatmap/marker view with the segment view, since the returned data of the ajax call is different. 

For heatmap/marker view, the accident is returned, which is placed at the acc_list.

While for the segment view, the segment is returned, which should be placed at another variable.